### DISCUSSION TITLE
about V2.0.0 and V1.3.3. Successly compress model but program goes wrong while running lammps

### DISCUSSION THREAD

**Discussion by zhao-w-en** [deepmodeling/deepmd-kit#1232] [Q&A]
Dear all:
    Recently I have install **V2.0.0** and try to **run lammps** with compressed model. At first I used **uncompressed model** like  
`pair_style	deepmd frozen_model.pb`
`pair_coeff	* *`
, but when I submit the lammps task I met error like this
`Invalid argument: Tensor model_attr/model_version:0, specified in either feed_devices or fetch_devices was not found in the Graph`

`Invalid argument: Tensor model_attr/model_version:0, specified in either feed_devices or fetch_devices was not found in the GraphInvalid argument: Tensor model_attr/model_version:0, specified in either feed_devices or fetch_devices was not found in the Graph`

`APPLICATION TERMINATED WITH THE EXIT STRING: Aborted (signal 6)`

`terminate called after throwing an instance of 'std::runtime_error'  what():  incompatable model: version 0.0 in graph, but version 1.0  supported `

`terminate called after throwing an instance of 'terminate called after throwing an instance of 'std::runtime_error'`

Then I **changed** the in.lammps as
 `pair_style	deepmd frozen_model_compress.pb`
`pair_coeff	* *`
and the error changed like this
`Internal: 2 root error(s) found.`
`  (0) Internal: Blas xGEMMStridedBatched launch failed : a.shape=[14,4,200], b.shape=[14,4,4], m=200, n=4, k=4, batch_size=14`
	` [[{{node filter_type_1/MatMul}}]]`
	` [[o_force/_31]]`
 ` (1) Internal: Blas xGEMMStridedBatched launch failed : a.shape=[14,4,200], b.shape=[14,4,4], m=200, n=4, k=4, batch_size=14`
	` [[{{node filter_type_1/MatMul}}]]`
`0 successful operations.`
`0 derived errors ignored.`
I have no idea about these error, by the way, lammps will work properly use **v1.3.3** if I use uncompressed dp model. I'm really confusing. Could you please give me some advice?
Waitting for your reply

**Comment by WU2JING** [comment]
Hello, you can try to run lammps with V2.0.0. The compressed model suit for V2.0.0.

**Reply by zhao-w-en** [reply]
Thank you so much!

**Comment by ZCY10410** [comment]
Hi, do you fix this problem successfully? I meet the same problem when I run lammps with V2.0.0. I am testing a deep potential trained in published work. I guess the potential I test is trained with DeepMD-kit version 1.

Waiting for your reply.

**Comment by JTaozhang** [comment]
I meet the same problem, "INVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the GraphINVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph". My deepmd-kit version is v2.2.2, lmp is 23 Jun 2022.

**Reply by njzjz** [reply]
This model is likely generated by an old DeePMD-kit version. Are you sure the version is v2.2.2?

**Reply by JTaozhang** [reply]
yes, I am sure. I listed  the initial part of log file here. LAMMPS (23 Jun 2022 - Update 1)
  using 1 OpenMP thread(s) per MPI task
Loaded 1 plugins from /share/home/zhangtao/anaconda3/envs/deepmd2/lib/deepmd_lmp
Reading data file ...
  triclinic box = (0 0 0) to (23.52804 20.37588 25) with tilt (-11.76402 1.5e-15 2.7e-15)
WARNING: Triclinic box skew is large (src/domain.cpp:224)
  8 by 14 by 1 MPI processor grid
  reading atoms ...
  364 atoms
Finding 1-2 1-3 1-4 neighbors ...
  special bond factors lj:    0        0        0       
  special bond factors coul:  0        0        0       
     0 = max # of 1-2 neighbors
     0 = max # of 1-3 neighbors
     0 = max # of 1-4 neighbors
     1 = max # of special neighbors
  special bonds CPU = 0.015 seconds
  read_data CPU = 0.041 seconds
Summary of lammps deepmd module ...
  >>> Info of deepmd-kit:
  installed to:       /share/home/zhangtao/anaconda3/envs/deepmd2
  source:             v2.2.2
  source branch:       HEAD
  source commit:      92ca097b
  source commit at:   2023-05-24 13:45:03 +0800
  surpport model ver.:1.1 
  build variant:      cpu
  build with tf inc:  /share/home/zhangtao/anaconda3/envs/deepmd2/include;/share/home/zhangtao/anaconda3/envs/deepmd2/include
  build with tf lib:  /share/home/zhangtao/anaconda3/envs/deepmd2/lib/libtensorflow_cc.so
  set tf intra_op_parallelism_threads: 0
  set tf inter_op_parallelism_threads: 0
  >>> Info of lammps module:
  use deepmd-kit at:  /share/home/zhangtao/anaconda3/envs/deepmd2INVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph
INVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph
INVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph
INVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph
INVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph
INVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph

**Reply by njzjz** [reply]
I see very little information as it does not show the version of the Python code. Did this message stop the program?

**Comment by JTaozhang** [comment]
Hi,
the output file doesn't contain any python information. I just run lammps. I don't think it has any connection with python.  The message didn't stop the program. If you assign the program with 56 cores cpu, the messsage will repeat 56 times.

**the start of the message like this,**
LAMMPS (23 Jun 2022 - Update 1)
  using 1 OpenMP thread(s) per MPI task
Loaded 1 plugins from /share/home/zhangtao/anaconda3/envs/deepmd2/lib/deepmd_lmp
Reading data file ...
  triclinic box = (0 0 0) to (23.52804 20.37588 25) with tilt (-11.76402 1.5e-15 2.7e-15)
WARNING: Triclinic box skew is large (src/domain.cpp:224)
  8 by 14 by 1 MPI processor grid
  reading atoms ...
  364 atoms
Finding 1-2 1-3 1-4 neighbors ...
  special bond factors lj:    0        0        0       
  special bond factors coul:  0        0        0       
     0 = max # of 1-2 neighbors
     0 = max # of 1-3 neighbors
     0 = max # of 1-4 neighbors
     1 = max # of special neighbors
  special bonds CPU = 0.016 seconds
  read_data CPU = 0.045 seconds
Summary of lammps deepmd module ...
  >>> Info of deepmd-kit:
  installed to:       /share/home/zhangtao/anaconda3/envs/deepmd2
  source:             v2.2.2
  source branch:       HEAD
  source commit:      92ca097b
  source commit at:   2023-05-24 13:45:03 +0800
  surpport model ver.:1.1 
  build variant:      cpu
  build with tf inc:  /share/home/zhangtao/anaconda3/envs/deepmd2/include;/share/home/zhangtao/anaconda3/envs/deepmd2/include
  build with tf lib:  /share/home/zhangtao/anaconda3/envs/deepmd2/lib/libtensorflow_cc.so
  set tf intra_op_parallelism_threads: 0
  set tf inter_op_parallelism_threads: 0
  >>> Info of lammps module:
  use deepmd-kit at:  /share/home/zhangtao/anaconda3/envs/deepmd2INVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph
INVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph

**the end of the message like this,**
NVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph
INVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph
INVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph
INVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph
INVALID_ARGUMENT: Tensor spin_attr/ntypes_spin:0, specified in either feed_devices or fetch_devices was not found in the Graph
  >>> Info of model(s):
  using   1 model(s): graph.001.pb 
  rcut in model:      7
  ntypes in model:    1

CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE

Your simulation uses code contributions which should be cited:
- USER-DEEPMD package:
The log file lists these citations in BibTeX format.

CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE

WARNING: Using 'neigh_modify every 1 delay 0 check yes' setting during minimization (src/min.cpp:187)
Generated 0 of 0 mixed pair_coeff terms from geometric mixing rule
Neighbor list info ...
  update every 1 steps, delay 0 steps, check yes
  max neighbors/atom: 4000, page size: 200000
  master list distance cutoff = 7.3
  ghost atom cutoff = 7.3
  binsize = 3.65, bins = 10 6 7
  1 neighbor lists, perpetual/occasional/extra = 1 0 0
  (1) pair deepmd, perpetual

Although the program report the message, the program still can be finished normally. But it is annoying to see these message.

**Reply by njzjz** [reply]
The Python interface generates the model, and it is essential to check if the Python and C++ interface versions match.

**Reply by JTaozhang** [reply]
Hi,

So, how to check the match between Python and C++ interface versions. and how can I solve this?  I install deepmd-kit through conda command directly, I think the interface of Python and C++ woubld be matched automatically. can you give me some advice to solve this ?

best regards,

**Reply by njzjz** [reply]
The version is printed on the screen during training.

**Comment by JTaozhang** [comment]
Yes, I have seen the python version is 3.10. I also listed the lib version here.

![image](https://github.com/deepmodeling/deepmd-kit/assets/125844761/37ba5588-0a67-4c26-a2ec-a8fc43874b74)

Do those lib versions confict with the python ?
